randovania/games/cave_story/exporter/patch_data_factory.py

```diff
@@ -158,24 +215,79 @@ def create_game_specific_data(self) -> dict:
+                starting_msg += (
+                    f"<GIT1006Got a =Life Capsule=!<ML+{num_to_tsc_value(self.life).decode('utf-8')}\r\n"
+                    f"Max health increased by\r\n"
+                    f"{self.life}!<WAI0010<NOD\r\n<CLR"
+                )
+
+            if starting_msg:
+                starting_msg += "<GIT0000\r\n"
+
+            if self.trades["blade"] >= 2:
+                starting_msg += (
+                    "You may trade the =Nemesis=\r\n"
+                    "with the =Blade= and vice-versa\r\n"
+                    "at the computer in Arthur's House.<WAI0025<NOD<FL+2811\r\n<CLR"
+                )
+
+            if self.trades["fireball"] >= 2:
+                starting_msg += (
+                    "You may trade the =Fireball=\r\n"
+                    "with the =Snake= and vice-versa\r\n"
+                    "at the computer in Arthur's House.<WAI0025<NOD<FL+2802\r\n<CLR"
+                )
+
+            # Consolidation items
+            if self.trades["keys"] >= 2:
+                starting_msg += "<IT+0040"
+            if self.trades["medals"] >= 2:
+                starting_msg += "<IT+0041"
+            if self.trades["lewd"] >= 2:
+                starting_msg += "<IT+0042"
+            if self.trades["sprinklers"] >= 2:
+                starting_msg += "<IT+0043"
+            if self.trades["mushrooms"] >= 2:
+                starting_msg += "<IT+0044"
+
+        starting_script = ""
+        if starting_msg:
+            starting_script += f"\r\n<PRI<MSG<TUR{starting_msg}<CLO"
+
+        starting_script += f"<EQ+{num_to_tsc_value(self.equip_num).decode('utf-8')}\r\n"
 
-        starting_items = self.patches.starting_resources()
+        # Starting HP
+        if self.configuration.starting_hp != 3 or self.life > 0:
+            starting_script += f"<ML+{num_to_tsc_value(self.configuration.starting_hp + self.life - 3).decode('utf-8')}"
 
+        return starting_script
+
+    def _grant_starting_items(self) -> str:
         starting_msg = ""
-        missile = next(
-            (res for res, _ in starting_items.as_resource_gain() if res.short_name in {"missile", "tempMissile"}), None
+        self.missile = next(
+            (res for res, _ in self.starting_items.as_resource_gain() if res.short_name in {"missile", "tempMissile"}),
+            None,
         )
-        for item, _ in starting_items.as_resource_gain():
-            if item.resource_type != ResourceType.ITEM or item == missile:
+        for item, _ in self.starting_items.as_resource_gain():
+            if item.resource_type != ResourceType.ITEM or item == self.missile:
                 continue
 
             if item.short_name == "lifeCapsule":
-                life = starting_items[item]
+                self.life = self.starting_items[item]
                 continue
 
             if item.short_name == "puppies":
-                num_puppies = starting_items[item]
+                num_puppies = self.starting_items[item]
```

Comment:
why do these variables need to be self?

Reply:
because they're used in two functions

Reply:
Ah hm. Well then.

Reply:
can you then at least define them under class level, rather than in a function?  Feels easier for me to follow.